source/h1_connection.c

```diff
@@ -880,26 +861,17 @@ static void s_outgoing_stream_task(struct aws_channel_task *task, void *arg, enu
         goto error;
     }
 
+    /* Set up callback so we can send another message when this one completes */
+    msg->on_completion = s_on_channel_write_complete;
+    msg->user_data = connection;
+
     /**
-     * Fill message with as much data as possible before sending.
-     * At first, we might be resuming work on a stream from a previous run of this task.
-     * Loop until no more streams have data to send,
-     * OR a stream still is unable to continue writing to the msg (probably because msg is full).
+     * Fill message data from the outgoing stream.
+     * Note that we might be resuming work on a stream from a previous run of this task.
      */
-    struct aws_h1_stream *outgoing_stream;
-    while ((outgoing_stream = s_update_outgoing_stream_ptr(connection)) != NULL) {
-        if (aws_h1_encoder_process(&connection->thread_data.encoder, &msg->message_data)) {
-            /* Error sending data, abandon ship */
-            goto error;
-        }
-
-        /* If there is a stream in progress, it means msg filled up before we finished a stream */
-        if (aws_h1_encoder_is_message_in_progress(&connection->thread_data.encoder)) {
-            break;
-        }
-
-        /* If stream is done sending data, mark as done sending, loop, and start sending the next stream's data */
-        outgoing_stream->is_outgoing_message_done = true;
+    if (aws_h1_encoder_process(&connection->thread_data.encoder, &msg->message_data)) {
```

Comment:
just checking that aws_h1_encoder_process raises an error in this case, otherwise we need to raise here.

Reply:
yep